x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/sync_integrations_on_remote.ts

```diff
@@ -180,3 +220,224 @@ export const syncIntegrationsOnRemote = async (
+  ccrIntegrations: IntegrationsData[],
+  installedIntegrationsByName: Record<string, SavedObjectsFindResult<Installation>>
+) => {
+  const integrationsStatus: RemoteSyncedIntegrationsStatus[] | undefined = ccrIntegrations?.map(
+    (ccrIntegration) => {
+      const localIntegrationSO = installedIntegrationsByName[ccrIntegration.package_name];
+      if (!localIntegrationSO) {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Installation not found on remote`,
+        };
+      }
+      if (ccrIntegration.package_version !== localIntegrationSO?.attributes.version) {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Found incorrect installed version ${localIntegrationSO?.attributes.version}`,
+        };
+      }
+      if (localIntegrationSO?.attributes.install_status !== 'installed') {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Installation status: ${localIntegrationSO?.attributes.install_status}`,
+        };
+      }
+      return {
+        ...ccrIntegration,
+        sync_status: 'completed' as SyncStatus.COMPLETED,
+        updated_at: localIntegrationSO?.updated_at,
+      };
+    }
+  );
+  return { integrations: integrationsStatus };
+};
+
+const compareCustomAssets = async (
+  esClient: ElasticsearchClient,
+  logger: Logger,
+  ccrCustomAssets: { [key: string]: CustomAssetsData }
+) => {
+  if (!ccrCustomAssets) return;
+
+  const abortController = new AbortController();
+
+  try {
+    const installedPipelines = await getPipeline(esClient, CUSTOM_ASSETS_PREFIX, abortController);
+    const installedComponentTemplates = await getComponentTemplate(
+      esClient,
+      CUSTOM_ASSETS_PREFIX,
+      abortController
+    );
+
+    const componentTemplatesByName = (
+      installedComponentTemplates?.component_templates || []
+    ).reduce((acc, componentTemplate) => {
+      if (componentTemplate?.name) {
+        acc[componentTemplate.name] = componentTemplate.component_template?.template;
+      }
+      return acc;
+    }, {} as Record<string, ClusterComponentTemplateSummary>);
+    const result: RemoteSyncedCustomAssetsRecord = {};
+
+    // compare custom pipelines and custom component templates
+    Object.entries(ccrCustomAssets).forEach(([ccrCustomName, ccrCustomAsset]) => {
+      if (ccrCustomAsset.type === 'ingest_pipeline') {
+        const installedAsset = installedPipelines[ccrCustomAsset?.name];
+
+        if (isEqual(installedAsset, ccrCustomAsset?.pipeline)) {
+          result[ccrCustomName] = {
+            ...ccrCustomAsset,
+            sync_status: 'completed' as SyncStatus.COMPLETED,
+          } as RemoteSyncedCustomAssetsStatus;
+        } else {
+          result[ccrCustomName] = {
+            ...ccrCustomAsset,
+            sync_status: 'failed' as SyncStatus.FAILED,
```

Comment:
Should this be `SYNCHRONIZING` state instead? I think for `FAILED` we would need an error.

Reply:
I'm not totally sure about this. In the `if` statement I'm checking the deep equality between the two pipelines assets, so in the `else` I marked the sync as `failed` because they're not the same. I think there are two options:

1. This should be `SYNCHRONIZING` and in that case it would stay in this statet until the next execution of the task. Do you think it's acceptable?
2. I leave it as `FAILED` but add an error

In general I haven't add the `SYNCHRONIZING` state anywhere, shouldn't it really be on the sync task? If I add it in the comparison logic it doesn't really reflect the sync state of the integrations. What do you think?

Reply:
My thinking was that if the contents don't match, it probably means that the sync hasn't happened yet, that's why the `SYNCHRONIZING` state. 
The other possibility is that the sync failed, but we don't have a way of knowing that, as we don't store the sync error anywhere for custom assets . I think we need a place to store these errors to be able to query in the status API. We can probably create a separate issue to store the errors and report in status API.

Reply:
> My thinking was that if the contents don't match, it probably means that the sync hasn't happened yet, that's why the SYNCHRONIZING state.

Ok it makes sense, checking the deep equality doesn't give us much context on what's happening.

>  I think we need a place to store these errors to be able to query in the status API. We can probably create a separate issue to store the errors and report in status API.

That's what I was thinking as well, a `status` field could be added directly in the index and kept updated by the task. I'll see if it's easy to add in this PR otherwise I can create a separate task.